packages/workspace/src/task-runner/default-task-runner.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  BuildEvent,
+  BuildEventType,
+  Task,
+  TaskCompleteEvent,
+  TasksRunner
+} from './task-runner';
+import * as runAll from 'npm-run-all';
+import { Observable } from 'rxjs';
+import { cliCommand } from '../command-line/shared';
+import { basename } from 'path';
+import * as fs from 'fs';
+import { output } from '@nrwl/workspace/src/command-line/output';
+
+const commonCommands = ['build', 'test', 'lint', 'e2e', 'deploy'];
+
+export interface DefaultTasksRunnerOptions {
+  parallel: boolean;
+  maxParallel: number;
+}
+
+export const defaultTasksRunner: TasksRunner = (
+  tasks: Task[],
+  options: DefaultTasksRunnerOptions
+): Observable<TaskCompleteEvent> => {
+  const additionalTaskOverrides = getLegacyTaskOverrides(options);
+  tasks.forEach(task => {
+    task.overrides = {
+      ...task.overrides,
+      ...additionalTaskOverrides
+    };
+  });
+  return new Observable(subscriber => {
+    const commands = getCommands(tasks);
+    runAll(commands, {
+      parallel: options.parallel || false,
+      maxParallel: options.maxParallel || 1,
+      continueOnError: true,
+      stdin: process.stdin,
+      stdout: process.stdout,
+      stderr: process.stderr
+    })
+      .then(() => {
+        tasks.forEach(task => {
+          subscriber.next({
+            task,
+            type: BuildEventType.TaskComplete,
+            success: true
+          });
+        });
+      })
+      .catch(e => {
+        e.results.forEach((result, i) => {
+          subscriber.next({
+            task: tasks[i],
+            type: BuildEventType.TaskComplete,
+            success: result.code === 0
+          });
+        });
+      })
+      .finally(() => {
+        subscriber.complete();
+        // fix for https://github.com/nrwl/nx/issues/1666
+        if (process.stdin['unref']) (process.stdin as any).unref();
+      });
+  });
+};
+
+function getLegacyTaskOverrides(options: any) {
+  const legacyTaskOverrides = { ...options };
+  delete legacyTaskOverrides.maxParallel;
```

Comment:
the other options (e.g., --head) are filtered out elsewhere. right?

Reply:
Yup, they are filtered out within `affected.ts#processArgs`